### PR TITLE
[None][perf] make perfect router rank-aware for EP MoE

### PR DESCRIPTION
**Pull request by bobboli**
## Summary
- make Renormalize perfect-router logits rank-aware by offsetting the EP routing schedule with `ep_rank`
- thread `mapping.moe_ep_rank` through the cached perfect-router helpers used by GPT-OSS and DeepSeek-V3 MoE paths
- add a routing unit test that checks the generated expert assignments match the `bench_moe_comm.py --perfect_router` schedule

## Testing
- `python -m py_compile tensorrt_llm/_torch/modules/fused_moe/routing.py tensorrt_llm/_torch/models/modeling_gpt_oss.py tensorrt_llm/_torch/models/modeling_deepseekv3.py tests/unittest/_torch/modules/test_moe_routing.py`
- `PYTHONPATH=. pytest -q tests/unittest/_torch/modules/test_moe_routing.py -k "rank_aware_perfect_router or renormalize_expert_load_balanced_logits"`

## Notes
- this PR intentionally does not include benchmark scripts or result artifacts

## Release Notes

* **New Features**
  * Enhanced Mixture of Experts (MoE) routing with rank-aware expert assignment and improved load balancing calculations across expert partitions.
  * Extended logits caching to incorporate partition-specific routing context for optimized performance.

* **Tests**
  * Added comprehensive test coverage for rank-aware expert routing scenarios.

### CODE REVIEW

**Comment by xxi-nv** [comment]
I believe that we can drop this PR, considering I have fix this issue in https://github.com/NVIDIA/TensorRT-LLM/pull/13250.
